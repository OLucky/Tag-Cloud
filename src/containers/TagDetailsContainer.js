import {connect} from "react-redux";

import TagDetails from "../components/TagDetails/TagDetails";

import {getTagDetailsData} from "../store/selectors";

const mapStateToProps = (state, props) => {
	return({
		tag: getTagDetailsData(state, {id: props.match.params.id}),
		settings: state.settings
	});
};

export default connect(mapStateToProps)(TagDetails);