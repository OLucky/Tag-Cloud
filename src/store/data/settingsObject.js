export const settingsObject = {
	minFontSize: 16,
	maxFontSize: 160,
	colors: ["#000000", "#0000FF", "#FF00FF",
		"#008000", "#800000",
		"#000080", "#808000", "#800080",
		"#C0C0C0", "#008080", "#FF6347"],
	numberOfPages: 9
};