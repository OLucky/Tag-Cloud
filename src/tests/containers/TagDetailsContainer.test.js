import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {createMockStore} from "redux-test-utils";
import {StaticRouter} from "react-router-dom";

import "../config";

import TagDetailsContainer from "../../containers/TagDetailsContainer";
import {tagsArray} from "../../store/data/tagsArray";
import {settingsObject} from "../../store/data/settingsObject";


const oneTag = tagsArray[0];

const testState = {
	tags: [oneTag],
	settings: settingsObject
};

describe("TagDetails Container", () => {

	test("renders with children", () => {
		const mockStore = createMockStore(testState);
		const mockMatch = {params: {id: oneTag.id}};
		const context = {};

		const component = shallow(
			<StaticRouter context={context}>
				<TagDetailsContainer match={mockMatch}/>
			</StaticRouter>, 
			{context: {store: mockStore}});

		expect(toJson(component)).toMatchSnapshot();
	});

	test("has correct props", () => {
		const mockStore = createMockStore(testState);
		const mockMatch = {params: {id: oneTag.id}};
		const context = {};

		const component = shallow(
			<StaticRouter context={context}>
				<TagDetailsContainer match={mockMatch}/>
			</StaticRouter>, 
			{context: {store: mockStore}});

		expect(component.find(TagDetailsContainer).dive().props().tag).toEqual({
			label: oneTag.label,
			id: oneTag.id,
			sentiment: oneTag.sentiment,
			pageType: oneTag.pageType,
			volume: oneTag.volume
		});	
		expect(component.find(TagDetailsContainer).dive().props().settings).toEqual(settingsObject);	
	});

});