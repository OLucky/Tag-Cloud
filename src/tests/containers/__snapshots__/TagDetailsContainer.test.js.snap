// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagDetails Container renders with children 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(TagDetails)
    match={
      Object {
        "params": Object {
          "id": "1751295897__Odessa",
        },
      }
    }
  />
</Router>
`;
